validator/src/cli.rs

```diff
@@ -293,6 +293,15 @@ pub fn app<'a>(version: &'a str, default_args: &'a DefaultArgs) -> App<'a, 'a> {
                 .takes_value(true)
                 .help("Use DIR as separate location for incremental snapshot archives [default: --snapshots value]"),
         )
+        .arg(
+            Arg::with_name("snapshot_from_dir")
+            .long("snapshot-from-dir")
```

Comment:
sounds good.  I would prefer boot-from-local-state.  

fastboot looks too generic.  if there is another piece of boot optimization work, it will not be clear which fastboot really selects.

Reply:
i think in the future we can set it up this way: allow the users choose to start from some archive.  if not chosen, the default is to boot from local state.  Handling that option, and unarchiving an archive to a snapshot dir if needed, can be done in a preprocessing tool which covers devops work.  Let the main validator image focus on main validator business.  In that way, the validator image always start from a snapshot dir, so there be no such arg for it.

Reply:
If I were a user, I would not want to know I have the option to choose fastboot or slowboot.  I obviously just want to be fast as the implementation allows, never choose to be slow.  I may at some time have doubt about the existing local state, and want to start from a trusted public archive.  So, I think the default really should just be from local state, but let the user choose to get an archive.  Typically users know that downloading and unarchiving an archive take time, so implicitly know that route will be slow.  Our user interface does not have to educate them on that.  So, I think the most concise way is to have the default be from the local state, and give the user an option to boot from an external archive. 

That is for the long term.  For now, given that from-local-state is still at the early experimental stage, it is right to enable it with an arg, not letting it be the default.

Reply:
update the arg.